src/tools/rbd/action/Bench.cc

```diff
@@ -293,18 +348,38 @@ int execute(const po::variables_map &vm) {
     return r;
   }
 
-  r = do_bench_write(image, bench_io_size, bench_io_threads, bench_bytes,
-                     bench_random);
+  r = do_bench(image, bench_io_type, bench_io_size, bench_io_threads,
+		     bench_bytes, bench_random);
   if (r < 0) {
-    std::cerr << "bench-write failed: " << cpp_strerror(r) << std::endl;
+    std::cerr << "do_bench failed: " << cpp_strerror(r) << std::endl;
     return r;
   }
   return 0;
 }
 
-Shell::Action action(
-  {"bench-write"}, {}, "Simple write benchmark.", "", &get_arguments, &execute, false);
+int execute_for_write(const po::variables_map &vm) {
+  return bench_execute(vm, IO_TYPE_WRITE);
```

Comment:
Since ```bench-write``` is hidden in the help menu now, it would probably be a good idea to output a warning that this is deprecated and to switch to "bench --io-type write".